hello-world/frontend/src/App.vue

```diff
@@ -1,40 +1,110 @@
+<template>
+   <div id="app" v-if="!mainLoading">
+    <h1>Greeter says: {{ greeting }} 👋</h1>
+    <div class="title">
+      This a simple dApp, which can choose fee token and interact with the
+      `Greeter` smart contract. 
+      <p>The contract is deployed on the zkSync testnet on <a :href="`https://sepolia.explorer.zksync.io/address/${GREETER_CONTRACT_ADDRESS}`" target="_blank">{{ GREETER_CONTRACT_ADDRESS }}</a></p>
+    </div>
+    <div class="main-box">
+      <div>
+        Select token:
+        <select v-model="selectedTokenAddress" @change="changeToken">
+          <option
+            v-for="token in tokens"
+            v-bind:value="token.address"
+            v-bind:key="token.address"
+          >
+            {{ token.symbol }}
+          </option>
+        </select>
+      </div>
+      <div class="balance" v-if="selectedToken">
+        <p>
+          Balance: <span v-if="retrievingBalance">Loading...</span>
+          <span v-else>{{ currentBalance }} {{ selectedToken.symbol }}</span>
+        </p>
+        <p>
+          Expected fee: <span v-if="retrievingFee">Loading...</span>
+          <span v-else>{{ currentFee }} {{ selectedToken.symbol }}</span>
+          <button class="refresh-button" @click="updateFee">Refresh</button>
+        </p>
+      </div>
+      <div class="greeting-input">
+        <input
+          v-model="newGreeting"
+          :disabled="!selectedToken || txStatus != 0"
+          placeholder="Write new greeting here..."
+        />
+
+        <button
+          class="change-button"
+          :disabled="!selectedToken || txStatus != 0 || retrievingFee"
+          @click="changeGreeting"
+        >
+          <span v-if="selectedToken && !txStatus">Change greeting</span>
+          <span v-else-if="!selectedToken">Select token to pay fee first</span>
+          <span v-else-if="txStatus === 1">Sending tx...</span>
+          <span v-else-if="txStatus === 2"
+            >Waiting until tx is committed...</span
+          >
+          <span v-else-if="txStatus === 3">Updating the page...</span>
+          <span v-else-if="retrievingFee">Updating the fee...</span>
+        </button>
+      </div>
+    </div>
+  </div>
+  <div id="app" v-else>
+    <div class="start-screen">
+      <h1>Welcome to Greeter dApp!</h1>
+      <button v-if="correctNetwork" @click="connectMetamask">Connect Metamask</button>
+      <button v-else @click="addZkSyncSepolia">Switch to zkSync Sepolia</button>
+
+    </div>
+  </div>
+</template>
+
 <script setup lang="ts" >
-import { ref, onMounted } from 'vue'
-// TODO: import ethers and zksync-ethers
+import { ref, onMounted } from "vue"
+import { Provider, Wallet, Contract, BrowserProvider } from "zksync-ethers";
```

Comment:
again, this is added in the tutorial, it should not be included yet

Reply:
I understand the one about package.json, but not sure what's wrong in this place

Reply:
we add the import in the tutorial as well. We should not include it

```suggestion
```